distributed/core.py

```diff
@@ -132,12 +132,12 @@ class Server(TCPServer):
     default_port = 0
 
     def __init__(self, handlers, max_buffer_size=MAX_BUFFER_SIZE,
-            connection_limit=512, **kwargs):
+            connection_limit=512, ssl_context_creator= create_ssl_context, **kwargs):
         self.handlers = assoc(handlers, 'identity', self.identity)
         self.id = str(uuid.uuid1())
         self._port = None
         self.rpc = ConnectionPool(limit=connection_limit)
-        super(Server, self).__init__(max_buffer_size=max_buffer_size, **kwargs)
+        super(Server, self).__init__(max_buffer_size=max_buffer_size, ssl_options=ssl_context_creator(), **kwargs)
```

Comment:
It could continue remaining separate from the Server object if we stick with the Tornado default of passing in `ssl_options` rather than a function that gives us these options.  It would just be on the caller of `Server` to call the function rather than sending the function-to-be-called to `Server`.

Alternatively, what is the benefit to passing around an options-creator rather than the options themselves?  What do we gain by delaying the function call?